### PR TITLE
Db improvements

### PR DESCRIPTION
**Pull request by andpii**
Need to be reviewed. Not sure that did implementation right. And also i will change README diagram after successfull review.
- Removed _Name_ property in _Task_ entity. Because _Summary_ property already exists.
- Added tables for _Task_ entity instead of enums.
- Added _UserProjectRole_ entity for linking _Role_ with specific _User_ and specific _Project_.
- Added field _Key_ to entity _Project_

### CODE REVIEW

**Comment by Anatolii-Ivanov** [comment]
Update readme scheme

**Comment by andpii** [comment]
> Update readme scheme

Done

File: backend/Tasque.Core.Common/Entities/Task.cs [hunk 1]
```diff
@@ -11,17 +10,21 @@ public Task()
...
-    public TaskState State { get; set; }
-    public TaskType Type { get; set; }
-    public TaskPriority Priority { get; set; }
+    public string? Description { get; set; }
 
     public DateTime Deadline { get; set; }
     public DateTime? FinishedAt { get; set; }
 
+    public int StateId { get; set; }
+    public TaskState State { get; set; } = null!;
```
Comment (Its-OP) [inline backend/Tasque.Core.Common/Entities/Task.cs:20]: State, Type, Priority. It's understandable that they are related to Task.

File: backend/Tasque.Core.Common/Entities/TaskPriority.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using Tasque.Core.Common.Entities.Abstract;
+
+namespace Tasque.Core.Common.Entities;
+
+public class TaskPriority : BaseEntity
```
Comment (Its-OP) [inline backend/Tasque.Core.Common/Entities/TaskPriority.cs:5]: Should be enum. We will not change them often. Same for State and Type.

**Review by Its-OP — CHANGES_REQUESTED** [review]

File: backend/Tasque.Core.Common/Entities/TaskPriority.cs (hunk 2 above)
Comment (andpii) [reply backend/Tasque.Core.Common/Entities/TaskPriority.cs:5]: Nope. It shouldn't be enum. I was told to made them as tables. So we can expand them.

File: backend/Tasque.Core.Common/Entities/Task.cs (hunk 1 above)
Comment (andpii) [reply backend/Tasque.Core.Common/Entities/Task.cs:20]: It's understandable when they are placed in one file. But if these files will be places in entities folder we won't be able to recognize to what do these models are related.

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by Its-OP — APPROVED** [review]

**Pull request merged**